test/uk/gov/hmrc/gform/formtemplate/JsonSchemeValidatorSpec.scala

```diff
@@ -0,0 +1,719 @@
+
+import io.circe.Json
+import io.circe.literal.JsonStringContext
+import io.circe.syntax.EncoderOps
+import munit.FunSuite
+import play.api.libs.json.{ JsArray, JsValue }
+import play.api.libs.json.Json.toJson
+import uk.gov.hmrc.gform.exceptions.SchemaValidationException
+
+class JsonSchemeValidatorSpec extends FunSuite {
+
+  private def constructTestOneSectionJsonTemplate(
+    propertyDependencies: Map[String, String],
+    strings: Map[String, String],
+    ints: Map[String, Int],
+    arraysOfStrings: Map[String, Array[String]],
+    arraysOfObjects: Map[String, Array[Map[String, String]]],
+    objects: Map[String, Map[String, String]]
+  ): Json = {
+    val basePropertiesAndValues = Map(
+      "id"    -> "TestID",
+      "label" -> "Test Label"
+    )
+
+    val allFields = basePropertiesAndValues.asJson
+      .deepMerge(propertyDependencies.asJson)
+      .deepMerge(strings.asJson)
+      .deepMerge(ints.asJson)
+      .deepMerge(arraysOfStrings.asJson)
+      .deepMerge(arraysOfObjects.asJson)
+      .deepMerge(objects.asJson)
+
+    json"""
+      {
+        "_id": "json-id",
+        "formName": "Json",
+        "version": 1,
+        "description": "",
+        "sections": [
+          {
+            "title": "Page",
+            "fields": [
+               $allFields
+            ]
+          }
+        ]
+      }"""
+  }
+
+  private def getNumberOfErrors(errorsAsList: List[JsValue]): Int =
+    // Get the number of custom conditional validation error messages
+    errorsAsList
+      .map(_.toString().contains("can only be used with"))
+      .count(_ == true)
+
+  private def getConditionalValidationErrors(errorsAsList: List[JsValue], numberOfErrors: Int): List[JsValue] =
+    // Only need last errors because conditional validation messages will appear at the bottom of the stack trace
+    errorsAsList.slice(errorsAsList.length - numberOfErrors, errorsAsList.length)
+
+  private def runInvalidJsonTest(result: Either[SchemaValidationException, Unit], expectedResult: List[String]): Unit =
+    result match {
+      case Right(value) => fail("No error was returned from schema validation:\n" + value)
+
+      case Left(errors) =>
+        val errorsAsList = errors.errors.as[JsArray].value.toList
+
+        val numberOfErrors = getNumberOfErrors(errorsAsList)
+        assertEquals(numberOfErrors, expectedResult.length)
+
+        val conditionalValidationErrors = getConditionalValidationErrors(errorsAsList, numberOfErrors)
+        assertEquals(conditionalValidationErrors, expectedResult.map(toJson(_)))
+    }
+
+  test(
+    "validateJson rejects the form gracefully when the property infoType is used and the type property is not [info]"
+  ) {
+    val jsonTemplate = constructTestOneSectionJsonTemplate(
+      propertyDependencies = Map("type" -> "text"),
+      strings = Map("infoType" -> "noformat"),
+      ints = Map(),
+      arraysOfStrings = Map(),
+      arraysOfObjects = Map(),
+      objects = Map()
```

Comment:
Can we replace all these parameters with actual json under test? Like:
```
json"""
  {
    "infoType": "noformat",
    "type": "text",
    "id": "testId",
    "label": "test label"
  }"""
```
That would make the tests really nice to read.